modules/dnn/src/opencl/batchnorm.cl

```diff
@@ -40,33 +40,34 @@
 //
 //M*/
 
-#define Dtype float
-#define Dtype4 float4
-#define Dtype8 float8
+#pragma OPENCL EXTENSION cl_khr_fp16 : enable
```

Comment:
Does this break kernel compilation if fp16 does not supported and we request "float" only?

Reply:
add #if defined (cl_khr_fp16) before using the macro